Source/Core/Core/IOS/Network/Socket.cpp

```diff
@@ -712,24 +709,99 @@ void WiiSockMan::Update()
+  [[maybe_unused]]
+	  s32 ret = poll(ufds.data(), nfds, 0);
+
+  std::chrono::time_point<std::chrono::high_resolution_clock> now = std::chrono::high_resolution_clock::now();
+  auto elapsed_d = std::chrono::duration_cast<std::chrono::milliseconds>(now - last_time);
+  auto elapsed = elapsed_d.count();
+  last_time = now;
+
+  for (auto& pcmd : pending_polls)
   {
-    for (auto& pair : WiiSockets)
+    if (pcmd.timeout > 0) // don't touch negative timeouts
     {
-      WiiSocket& sock = pair.second;
-      sock.Update(FD_ISSET(sock.fd, &read_fds) != 0, FD_ISSET(sock.fd, &write_fds) != 0,
-                  FD_ISSET(sock.fd, &except_fds) != 0);
+      if (pcmd.timeout < elapsed)
+        pcmd.timeout = 0;
+      else
+        pcmd.timeout -= elapsed;
     }
   }
-  else
+
+  int n = 0;
+  for (auto& elem : WiiSockets)
   {
-    for (auto& elem : WiiSockets)
-    {
-      elem.second.Update(false, false, false);
-    }
+    elem.second.Update(ufds[n].revents & POLLIN,
+                       ufds[n].revents & POLLOUT,
+                       ufds[n].revents & (POLLHUP | POLLERR));
+    n++;
   }
+
+  struct
+  {
+    int native;
+    int wii;
+  } mapping[] = {
+      {POLLRDNORM, 0x0001}, {POLLRDBAND, 0x0002}, {POLLPRI, 0x0004}, {POLLWRNORM, 0x0008},
+      {POLLWRBAND, 0x0010}, {POLLERR, 0x0020},    {POLLHUP, 0x0040}, {POLLNVAL, 0x0080},
+  };
+
+  auto mid =
+    std::partition(pending_polls.begin(),
+                   pending_polls.end(),
+                   [&](auto &pcmd) {
+                       if (pcmd.timeout == 0)
+                         return true;
+                       pcmd.trigger_count =
+                         std::count_if(pcmd.wii_fds.begin(),
+                                       pcmd.wii_fds.end(),
+                                       [&](auto &e) {
+                                         return
+                                             find_if(ufds.begin(),
+                                                     ufds.end(),
+                                                     [&](auto &u) {
+                                                         return e.fd == u.fd &&
+                                                             (e.events & u.revents);
+                                                     }) != ufds.end();
+                                         });
+                       return !!pcmd.trigger_count;
+                     });
+  ret = mid - pending_polls.begin();
+  finished_polls.assign(std::make_move_iterator(pending_polls.begin()),
+						std::make_move_iterator(mid));
+  pending_polls.erase(pending_polls.begin(), mid); 
+  std::move(pending_polls.begin(),
+            mid,
+            std::back_inserter(finished_polls));
+  std::for_each(finished_polls.begin(),
+                finished_polls.end(),
+                [&](auto &pcmd) {
+                  auto r = pcmd.timeout == 0 ? 0 : pcmd.trigger_count;
+                  r = WiiSockMan::GetNetErrorCode(r, "SO_POLL", false);
```

Comment:
This function should only be used against network functions and right after they returned otherwise the behaviour is undefined. Especially, since it sets the last error value and on top of that calls `WSAGetLastError` on Windows.

It seems unnecessary here as it's not an error value.